aws/aws.go

```diff
@@ -188,8 +189,9 @@ func V2ConfigFromURLParams(ctx context.Context, q url.Values) (awsv2.Config, err
 			customResolver := awsv2.EndpointResolverWithOptionsFunc(
 				func(service, region string, options ...interface{}) (awsv2.Endpoint, error) {
```

Comment:
Do we not need to use "service"?

Reply:
I don't think we need to pass the service parameter to any other function or add it to an object. 
the service parameter is the name of the service and can be used for custom handling https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/

Reply:
It's being used only to check if the custom endpoint should be use for specific service
```
customResolver := aws.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (aws.Endpoint, error) {
    if service == dynamodb.ServiceID && region == "us-west-2" {
        return aws.Endpoint{
            PartitionID:   "aws",
            URL:           "https://test.us-west-2.amazonaws.com",
            SigningRegion: "us-west-2",
            HostnameImmutable: true,
        }, nil
    }
    return aws.Endpoint{}, fmt.Errorf("unknown endpoint requested")
})

cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithEndpointResolverWithOptions(customResolver))

```